### PR TITLE
file_server prints errors in console

### PR DESCRIPTION
**Pull request by zhmushan**
cc @kitsonk 
close: #3465

### CODE REVIEW

File: std/http/file_server.ts [hunk 1]
```diff
@@ -142,20 +142,26 @@ async function serveDir(
...
-  } else {
-    return {
-      status: 500,
-      body: encoder.encode("Internal server error")
-    };
+  if (e instanceof DenoError) {
+    if (e.kind === ErrorKind.PermissionDenied) {
+      return {
+        status: 500,
+        body: encoder.encode(e.message)
```
Comment (nayeemrmn) [inline std/http/file_server.ts]: Rather than serve the message, you should log the error to the console near the `serveFallback()` call in `listenAndServe()`. That's what was suggested by @kitsonk.

Comment (zhmushan) [reply std/http/file_server.ts]: hmm...so I have reverted these commits

File: std/http/file_server.ts [hunk 2]
```diff
@@ -142,6 +142,8 @@ async function serveDir(
 }
 
 async function serveFallback(req: ServerRequest, e: Error): Promise<Response> {
+  console.error(e.message);
```
Comment (nayeemrmn) [inline std/http/file_server.ts]: I would have put this before the _call_ of `serveFallback()`:https://github.com/denoland/deno/blob/a66f1ed2b543eda2496f46924eb3446e386e6af1/std/http/file_server.ts#L302

Comment (zhmushan) [reply std/http/file_server.ts]: agree

File: std/http/file_server.ts [hunk 3]
```diff
@@ -6,7 +6,7 @@
 // TODO Add tests like these:
 // https://github.com/indexzero/http-server/blob/master/test/http-server-test.js
 
-const { ErrorKind, cwd, args, stat, readDir, open } = Deno;
+const { ErrorKind, DenoError, cwd, args, stat, readDir, open } = Deno;
```
Comment (nayeemrmn) [inline std/http/file_server.ts]: You can revert this change it seems.

Comment (zhmushan) [reply std/http/file_server.ts]: I used [here](https://github.com/denoland/deno/pull/3467/files#diff-b6647ec3bad62bae632704dd0c54e7ceR145)

**Comment by zhmushan** [comment]
why ci red?

**Comment by axetroy** [comment]
> why ci red?
> It seems to fail where I haven't modified.

Try to run CI again with empty commit

```bash
git commit -m "trigger ci" --allow-empty
```

File: std/http/file_server_test.ts [hunk 4]
```diff
@@ -79,3 +79,23 @@ test(async function serveFallback(): Promise<void> {
...
+  const deniedServer = Deno.run({
+    args: [Deno.execPath(), "run", "--allow-net", "http/file_server.ts"],
+    stdout: "null",
+    stderr: "piped"
+  });
+
+  const curl = Deno.run({
+    args: ["curl", "http://localhost:4500/"],
+    stdout: "null"
+  });
```
Comment (ry) [inline std/http/file_server_test.ts]: I don't think the tests should depend on curl. Can you use fetch instead?

Comment (zhmushan) [reply std/http/file_server_test.ts]: @ry done.

**Comment by ry** [comment]
Hi @zhmushan - there's a conflict, could you please merge master? Otherwise looks good. Thank you!

**Comment by zhmushan** [comment]
@ry done
